torch/utils/data/datapipes/iter/selecting.py

```diff
@@ -1,13 +1,29 @@
+import warnings
 from torch.utils.data import IterDataPipe, functional_datapipe, DataChunk
 from typing import Callable, TypeVar, Iterator, Optional, Tuple, Dict
 
-from .callable import MapperIterDataPipe
-
 T_co = TypeVar('T_co', covariant=True)
 
+try:
+    import dill
+
+    # XXX: By default, dill writes the Pickler dispatch table to inject its
+    # own logic there. This globally affects the behavior of the standard library
+    # pickler for any user who transitively depends on this module!
+    # Undo this extension to avoid altering the behavior of the pickler globally.
+    dill.extend(use_dill=False)
+    DILL_AVAILABLE = True
+except ImportError:
+    DILL_AVAILABLE = False
+
+# Default function to return each item directly
+# In order to keep datapipe picklable, eliminates the usage
+# of python lambda function
+def default_filter_fn(data):
+    return True
```

Comment:
Let's remove it here. We should expect user to provide a filter_fn

Reply:
Done.